### PR TITLE
Save Realm into context

### PR DESCRIPTION
**Pull request by yegle**
This allows the realm id to be picked up by various metrics.

### CODE REVIEW

File: pkg/controller/middleware/auth.go [hunk 1]
```diff
@@ -122,6 +122,18 @@ func RequireAuth(cacher cache.Cacher, authProvider auth.Provider, db *database.D
 				return
 			}
 
+			// Insert realm id to context so metrics can pick it up.
+			realmID := controller.RealmIDFromSession(session)
+			// This is inserted by realm selector, so it could be 0 if no realm
+			// has been selected.
+			if realmID != 0 {
+				realm, err := db.FindRealm(realmID)
```
Comment (mikehelmick) [inline pkg/controller/middleware/auth.go:130]: this should attempt to read the realm from the cache before going to DB

**Review by mikehelmick — CHANGES_REQUESTED** [review]

**Review by sethvargo — COMMENTED** [review]
This is already set by apikey.go. I think this belongs in membership.go around L85. We shouldn't need to lookup the realm if we do it there:

```
ctx = controller.WithRealm(ctx, membership.Realm)
```

File: pkg/controller/middleware/auth.go (hunk 1 above)
Comment (sethvargo) [reply pkg/controller/middleware/auth.go:130]: You're right, but I think this is the wrong place to do this entirely. We should set this in membership.go instead, where the realm already exists with no lookup required.

**Review by sethvargo — APPROVED** [review]
/lgtm

**Pull request merged**